newIDE/app/src/UI/SearchBar.js

```diff
@@ -96,141 +109,266 @@ const getStyles = (props: Props, state: State) => {
   };
 };
 
+const useAutocompleteStyles = makeStyles(
+  ({
+    palette: {
+      text: { primary },
+    },
+  }) => ({
+    groupLabel: { color: `${primary}${primary.length > 4 ? '88' : '8'}` },
+  })
+);
+
+export type SearchBarInterface = {|
+  focus: () => void,
+  blur: () => void,
+|};
+
 /**
  * Material design search bar,
  * inspired from https://github.com/TeamWertarbyte/material-ui-search-bar
  *
- * Customized to add optional tags button.
+ * Customized to add optional menu button and chips corresponding to tags.
  */
-export default class SearchBar extends React.PureComponent<Props, State> {
-  state = {
-    focus: false,
-    value: this.props.value,
-    active: false,
-  };
-  _textField = React.createRef<TextField>();
+const SearchBar = React.forwardRef<Props, SearchBarInterface>(
+  (
+    {
+      disabled,
+      placeholder,
+      onChange,
+      onRequestSearch,
+      style,
+      value: parentValue,
+      tagsHandler,
+      tags,
+      buildMenuTemplate,
+      helpPagePath,
+    },
+    ref
+  ) => {
+    React.useImperativeHandle(ref, () => ({
+      focus,
+      blur,
+    }));
+    const focus = () => {
+      if (textField.current) {
+        textField.current.focus();
+      }
+    };
+    const blur = () => {
+      if (textField.current) {
+        textField.current.blur();
+      }
+    };
 
-  componentWillReceiveProps(nextProps: Props) {
-    if (this.props.value !== nextProps.value) {
-      this.setState({ ...this.state, value: nextProps.value });
-    }
-  }
+    // This variable represents the content of the input (text field)
+    const [value, setValue] = React.useState<string>(parentValue);
+    // This variable represents the value of the autocomplete, used to
+    // highlight an option and to determine if an option is selectable, or
+    // if an event should be fired when an option is selected.
+    const [autocompleteValue, setAutocompleteValue] = React.useState<string>(
+      parentValue
+    );
 
-  focus = () => {
-    if (this._textField.current) {
-      this._textField.current.focus();
-    }
-  };
+    const textField = React.useRef<?TextField>(null);
 
-  blur = () => {
-    if (this._textField.current) {
-      this._textField.current.blur();
-    }
-  };
+    const styles = getStyles(value, disabled);
+    const autocompleteStyles = useAutocompleteStyles();
 
-  handleFocus = () => {
-    this.setState({ focus: true });
-  };
+    const changeValue = React.useCallback(
+      newValue => {
+        setValue(newValue || '');
+        onChange && onChange(newValue || '');
+      },
+      [onChange, setValue]
+    );
 
-  handleBlur = () => {
-    this.setState({ focus: false });
-    if (this.state.value.trim().length === 0) {
-      this.setState({ value: '' });
-    }
-  };
+    React.useEffect(
+      () => {
+        setValue(parentValue);
+      },
+      [parentValue]
+    );
+    React.useEffect(
```

Comment:
overall, I feel that each useEffect could use a small comment, explaining why it's there.
This component is quite complete and it can take time to understand what it does

Reply:
Will do